fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/SemiReorderRule.java

```diff
@@ -104,6 +104,9 @@ public boolean check(final OptExpression input, OptimizerContext context) {
                 } else if (!isProjectToColumnRef &&
                         newSemiOutputColumns.containsAll(entry.getValue().getUsedColumns())) {
                     semiExpression.put(entry.getKey(), entry.getValue());
+                } else if (!isProjectToColumnRef && leftChildInputColumns.containsAll(entry.getValue().getUsedColumns())) {
```

Comment:
newSemiOutputColumns may compute error, if it is right, there are no need to add this